app/models/account.rb

```diff
@@ -33,6 +33,15 @@ def transactions
   end
 
   def accumulative_balance_data
-    transactions.group_by_day(:created_at).sum(:amount)
+    transactions_grouped_by_day = transactions.group_by { |t| t.created_at.to_date }
```

Comment:
This is not sorted by date, so the balance below is still not accumulating correctly